### PR TITLE
gitui: update to 0.10.1.

### PR DESCRIPTION
**Pull request by jcgruenhage**

### CODE REVIEW

File: srcpkgs/gitui/template [hunk 1]
```diff
@@ -1,14 +1,15 @@
 # Template file for 'gitui'
 pkgname=gitui
-version=0.9.1
+version=0.10.1
 revision=1
 build_style=cargo
+hostmakedepends="python3"
```
Comment (jcgruenhage) [inline srcpkgs/gitui/template:6]: A new dependency of gitui requires `python3` for building, which is what caused the previous CI failure.

**Comment by jcgruenhage** [comment]
Now it fails because it needs a newer rust version.

**Comment by ericonr** [comment]
Please try to test building locally. You should have tested the resulting package in runtime as well.

**Comment by jcgruenhage** [comment]
Isn't that what the CI is for?

**Comment by Duncaen** [comment]
> Isn't that what the CI is for?

CI is limited and it blocks other PRs.

**Comment by ericonr** [comment]
> Isn't that what the CI is for?

CI doesn't run package checks, so it's possible that one of the many moving pieces involved broke a new release. And even if it did run, checks can't cover all runtime related stuff. At least launching it quickly and being sure that basic functionality is still there is very important.

This isn't a critical package, so breakages are a smaller issue, but it isn't nice anyway.

**Comment by jcgruenhage** [comment]
> CI doesn't run package checks, so it's possible that one of the many moving pieces involved broke a new release. And even if it did run, checks can't cover all runtime related stuff. At least launching it quickly and being sure that basic functionality is still there is very important.

Luckily that's quite rare with rust (which is the reason I didn't do that stuff for version bumps so far, because I thought the risk was low enough), but I agree. Will compile locally and launch it at least once.

> This isn't a critical package, so breakages are a smaller issue, but it isn't nice anyway.

Of course, breakage is never nice.

> CI is limited and it blocks other PRs.

Well, so is my time and resources. waiting a bit until CI has run through doesn't block the workflow usually, building the packages locally is more overhead IMO.

Would integrating sccache into rust packages in xbps-src be accepted upstream? That would make it easier for me to test the packages locally. Would be optional of course, behind an env var or something like that.

**Comment by ericonr** [comment]
> Would integrating sccache into rust packages in xbps-src be accepted upstream? That would make it easier for me to test the packages locally. Would be optional of course, behind an env var or something like that.

I believe it would be very appreciated, but AFAIK it currently isn't possible, since they require a server setup and some other stuff.

**Comment by Duncaen** [comment]
You can also enable travis on your fork (for free) and run CI on branches you push to your fork before opening PRs, this wouldn't block the CI for void-linux/void-packages.

**Pull request merged**